tests/as_python_test.py

```diff
@@ -30,3 +57,73 @@ def test_complex(self):
         self.check_as_python(
             [b"", b"1", b"2", [b"30", b"40", b"90"], b"600", (b"", b"18")]
         )
+
+    def test_listp(self):
+        self.assertEqual(SExp.to(42).listp(), False);
+        self.assertEqual(SExp.to(b"").listp(), False);
+        self.assertEqual(SExp.to(b"1337").listp(), False);
+
+        self.assertEqual(SExp.to((1337, 42)).listp(), True);
+        self.assertEqual(SExp.to([1337, 42]).listp(), True);
+
+    def test_nullp(self):
+        self.assertEqual(SExp.to(b"").nullp(), True)
+        self.assertEqual(SExp.to(b"1337").nullp(), False)
+        self.assertEqual(SExp.to((b"", b"")).nullp(), False)
+
+    def test_constants(self):
+        self.assertEqual(SExp.__null__.nullp(), True)
+        self.assertEqual(SExp.null().nullp(), True)
+        self.assertEqual(SExp.true, True)
+        self.assertEqual(SExp.false, False)
+
+    def test_list_len(self):
+        v = SExp.to(42);
+        for i in range(100):
+            self.assertEqual(v.list_len(), i)
+            v = SExp.to((42, v))
+        self.assertEqual(v.list_len(), 100)
+
+    def test_as_int(self):
+        self.assertEqual(SExp.to(b'\x80').as_int(), -128)
+        self.assertEqual(SExp.to(b'\xff').as_int(), -1)
+        self.assertEqual(SExp.to(b'\x00\x80').as_int(), 128)
+        self.assertEqual(SExp.to(b'\x00\xff').as_int(), 255)
+
+    def test_cons(self):
+        # list
+        self.assertEqual(SExp.to(b'\x01').cons(SExp.to(b'\x02').cons(SExp.null())).as_python(), [b'\x01', b'\x02'])
+        # cons-box of two values
+        self.assertEqual(SExp.to(b'\x01').cons(SExp.to(b'\x02').as_python()), (b'\x01', b'\x02'))
+
+    def test_deep_recursion(self):
+        d = [b"2"]
+        for i in range(480):
+            d = [d]
+#        self.check_as_python(d)
```

Comment:
this fails, and it's not trivial to fix. there's a recursive function that probably should be made non-recursive